src/kudu/tserver/tablet_server_options.h

```diff
@@ -73,6 +73,10 @@ struct TabletServerOptions : public kudu::server::ServerBaseOptions {
   std::function<void()> ldcb;
   bool disable_noop = false;
 
+  // This is to enable a fresh instance join the ring with logs from
+  // a certain opid and term.
+  bool log_bootstrap_on_first_run = false;
```

Comment:
How is this different from `is_first_run` maintained in server_base.h. Can this be piggy backed there?

Reply:
is_first_run is a kuduraft thing. it means this is a fresh instance. However a fresh instance with no consensus-metadata
but having raft logs is only a mysql/fb thing so I am gating this functionality. If this works well, we can remove this and make it default true or use the presence of a log_factory to gate this.